### PR TITLE
REDIS-36 Create request handler and rate-limit invocation handler

### PR DESCRIPTION
**Pull request by EngineersBox**
## Changes
* Created RateLimiter class with decorator driven invocation
* Created templated member methods for rate limiting
* Created config for Redis and RL configurations

## Ticket
<https://jkl-tl.atlassian.net/browse/REDIS-36>

### CODE REVIEW

File: modules/RateLimiter/config.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CONNECTION = {
+    "host_address": "127.0.0.1", # Redis DB IPv4 address
+    "port": 1234, # Redis DB port
+    "db": 0 # Redis DB ID
+}
+LIMITING = {
+    "key_prefix": "ratelimit:", # Prefix for rate limiter keys
+    "max_reqs": 30, # Maximum number of requests in the expiry period
+    "expiry": 1 # Expiry period for an IP address in seconds
+}
```
Comment (ariadev79) [inline modules/RateLimiter/config.py]: as the error says: newline needed. Unless it's a python thing that I was not aware

File: modules/RateLimiter/config.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CONNECTION = {
+    "host_address": "127.0.0.1", # Redis DB IPv4 address
+    "port": 1234, # Redis DB port
+    "db": 0 # Redis DB ID
+}
+LIMITING = {
+    "key_prefix": "ratelimit:", # Prefix for rate limiter keys
+    "max_reqs": 30, # Maximum number of requests in the expiry period
```
Comment (ariadev79) [inline modules/RateLimiter/config.py:9]: How is this number decided?

**Review by ariadev79 — COMMENTED** [review]
Some suggestions and questions. For those sections that need to do, maybe add a TODO on top of it as well for better navigation

File: modules/RateLimiter/config.py (hunk 1 above)
Comment (EngineersBox) [reply modules/RateLimiter/config.py]: Yep good catch, ill fix that

File: modules/RateLimiter/config.py (hunk 2 above)
Comment (EngineersBox) [reply modules/RateLimiter/config.py:9]: Arbitrary, will need to be decided at some point. I'll add a todo there and create a ticket to decide (probs through some mild use case investigation)

Comment (ariadev79) [reply modules/RateLimiter/config.py:9]: Remember to chase Rong for the documentation regarding IP :p

**Review by ariadev79 — APPROVED** [review]
Review addressed

File: modules/RateLimiter/config.py (hunk 2 above)
Comment (EngineersBox) [reply modules/RateLimiter/config.py:9]: Yep, indeed. Closing the loop on determining an acceptable value, have created an investigation ticket here: <https://jkl-tl.atlassian.net/browse/REDIS-37>

**Pull request merged**